File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectSCcsharp/angajatModel/validators/AngajatValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace proiectSCcsharp.domain.validators
{
    public class AngajatValidator : Validator<Angajat>
    {
        public AngajatValidator()
        {
        }

        public void validate(Angajat entity)
        {
            String errors = "";
            if (entity.name == "")
                errors += "invalid name\n";
            if (entity.username == "")
                errors += "invalid username\n";
            if (entity.password == "")
                errors += "invalid password\n";
            if (errors.Length > 0)
                throw new ValidationException(errors);
        }
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 3-proiect Server-Client Java/proiectServerClientJava/Domain/src/main/java/proiectServerClientJava/domain/validators/AngajatValidator.java
package proiectServerClientJava.domain.validators;

import proiectServerClientJava.domain.Angajat;

import java.util.Objects;

public class AngajatValidator implements Validator<Angajat> {
    private static final AngajatValidator instance = new AngajatValidator();

    public static AngajatValidator getInstance() {
        return instance;
    }

    public AngajatValidator() {
    }

    @Override
    public void validate(Angajat entity) throws ValidationException {
        String errors = "";
        if (Objects.equals(entity.getName(), ""))
            errors += "invalid name\n";
        if (Objects.equals(entity.getUsername(), ""))
            errors += "invalid username\n";
        if (Objects.equals(entity.getPassword(), ""))
            errors += "invalid password\n";
        if (errors.length() > 0)
            throw new ValidationException(errors);
    }
}